File: Data Structures & Algorithms/anagram-groups/submission-3.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        char_groups = {}
        for s in strs:
            counts = self.count_chars(s)
            cc_counts = ".".join(str(counts))
            if cc_counts not in char_groups:
                char_groups[cc_counts] = [s]
            else:
                char_groups[cc_counts].append(s)

        return [s for s in char_groups.values()]




    
    def count_chars(self, mystr):
        counts = [0] * 26
        for c in mystr:
            counts[ord(c) - ord("a")] += 1
        return counts
        

File: Data Structures & Algorithms/is-anagram/submission-3.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        vector<int> counts(26, 0);
        for (char c : s){
            int idx = c - 'a';
            counts[idx] += 1;
        }

        for (char c : t){
            int idx = c - 'a';
            counts[idx] -= 1;
        }
        for (int i = 0; i < 26; i++){
            if (counts[i] != 0){
                return false;
            }
        }
        return true;
        
    }
};
